alg/teca_damper.cxx

```diff
@@ -0,0 +1,217 @@
+{
+    this->set_number_of_input_connections(1);
+    this->set_number_of_output_ports(1);
+}
+
+// --------------------------------------------------------------------------
+teca_damper::~teca_damper()
+{}
+
+// --------------------------------------------------------------------------
+std::vector<std::string> teca_damper::get_damper_variables(
+    const teca_metadata &request)
+{
+    std::vector<std::string> damper_vars = this->damper_variables;
+    if (damper_vars.empty() &&
+        request.has("teca_damper::damper_variables"))
+            request.get("teca_damper::damper_variables", damper_vars);
+        
+    return damper_vars;
+}
+
+// --------------------------------------------------------------------------
+std::vector<teca_metadata> teca_damper::get_upstream_request(
+    unsigned int port,
+    const std::vector<teca_metadata> &input_md,
+    const teca_metadata &request)
+{
+#ifdef TECA_DEBUG
+    cerr << teca_parallel_id()
+        << "teca_damper::get_upstream_request" << endl;
+#endif
+    (void) port;
+    (void) input_md;
+
+    vector<teca_metadata> up_reqs;
+
+    // get the name of the array to request
+    std::vector<std::string> damper_vars = this->get_damper_variables(request);
+    if (damper_vars.empty())
+    {
+        TECA_ERROR("damper variables were not specified")
+        return up_reqs;
+    }
+
+    // pass the incoming request upstream, and
+    // add in what we need
+    teca_metadata req(request);
+    std::set<std::string> arrays;
+    if (req.has("arrays"))
+        req.get("arrays", arrays);
+    arrays.insert(damper_vars.begin(), damper_vars.end());
+
+    req.insert("arrays", arrays);
+
+    // send up
+    up_reqs.push_back(req);
+
+    return up_reqs;
+}
+
+// --------------------------------------------------------------------------
+const_p_teca_dataset teca_damper::execute(
+    unsigned int port,
+    const std::vector<const_p_teca_dataset> &input_data,
+    const teca_metadata &request)
+{
+#ifdef TECA_DEBUG
+    cerr << teca_parallel_id()
+        << "teca_damper::execute" << endl;
+#endif
+    (void)port;
+
+    // get the input
+    const_p_teca_cartesian_mesh in_mesh =
+        std::dynamic_pointer_cast<const teca_cartesian_mesh>(
+            input_data[0]);
+    if (!in_mesh)
+    {
+        TECA_ERROR("empty input, or not a mesh")
+        return nullptr;
+    }
+
+    // create output and copy metadata, coordinates, etc
+    p_teca_cartesian_mesh out_mesh =
+        std::dynamic_pointer_cast<teca_cartesian_mesh>(in_mesh->new_instance());
+
+    out_mesh->shallow_copy(
+        std::const_pointer_cast<teca_cartesian_mesh>(in_mesh));
+
+    // get the input array names
+    std::vector<std::string> damper_vars = this->get_damper_variables(request);
+    if (damper_vars.empty())
+    {
+        TECA_ERROR("A damper variable was not specified")
+        return nullptr;
+    }
+
+    // get threshold values
+    double filter_lat_width_val = this->filter_lat_width_value;
```

Comment:
I'd like to set this up so the user will specify width as a half width at half max (HWHM). for instance a setting of 10 deg lat would result in the damping factor of 0.5 at +/- 10 deg lat. The conversion from HWHM to sigma is `sigma = hwhm/(sqrt(2*ln(2)))` While this is arbitrary, imo it's more intuitive than using sigma directly.

Reply:
we will wait on this